explorer/homepage/index.tsx

```diff
@@ -11,92 +11,89 @@ import SettingsPage from '@components/settingspage';
-  constructor() {
-    super(null);
-    this.state = {
-      showHomePage: true,
-      showSettingsPage: false,
-      themes: ['Auto', 'Light', 'Dark'],
-      currentTheme: lynx.__globalProps.preferredTheme || 'Auto',
-    };
-  }
+export default function Explorer() {
+  const [state, setState] = useState<ContainerPageState>({
+    showHomePage: true,
+    showSettingsPage: false,
+    themes: ['Auto', 'Light', 'Dark'],
+    currentTheme: lynx.__globalProps.preferredTheme || 'Auto',
+  });
 
-  openHomePage = () => {
-    if (this.state.showHomePage) {
+  const openHomePage = () => {
+    if (state.showHomePage) {
       return;
     }
-    this.setState({
+    setState((prev) => ({
+      ...prev,
       showHomePage: true,
       showSettingsPage: false,
-    });
+    }));
   };
 
-  openSettingsPage = () => {
-    if (this.state.showSettingsPage) {
+  const openSettingsPage = () => {
+    if (state.showSettingsPage) {
       return;
     }
-    this.setState({
+    setState((prev) => ({
+      ...prev,
       showHomePage: false,
       showSettingsPage: true,
-    });
+    }));
   };
 
-  setTheme = (theme: string) => {
-    if (this.state.currentTheme == theme) {
+  const setTheme = (theme: string) => {
+    if (state.currentTheme === theme) {
       return;
     }
 
-    this.setState({
+    setState((prev) => ({
+      ...prev,
       currentTheme: theme,
-    });
+    }));
 
     NativeModules.ExplorerModule.saveThemePreferences('preferredTheme', theme);
   };
 
-  withTheme = (className: string) => {
-    const { currentTheme } = this.state;
+  const withTheme = (className: string) => {
+    const { currentTheme } = state;
     if (currentTheme !== 'Auto') {
       return `${className}__${currentTheme.toLowerCase()}`;
     }
     return `${className}__${lynx.__globalProps.theme.toLowerCase()}`;
   };
 
-  withNotchScreen = (className: string) => {
+  const withNotchScreen = (className: string) => {
     return lynx.__globalProps.isNotchScreen ? `${className}__notch` : className;
   };
 
-  render() {
-    return (
-      <view clip-radius="true" style="height: 100%">
-        <HomePage
-          currentTheme={this.state.currentTheme}
-          withNotchScreen={this.withNotchScreen}
-          withTheme={this.withTheme}
-          showPage={this.state.showHomePage}
-        />
-        <SettingsPage
-          themes={this.state.themes}
-          currentTheme={this.state.currentTheme}
-          withNotchScreen={this.withNotchScreen}
-          withTheme={this.withTheme}
-          setTheme={this.setTheme}
-          showPage={this.state.showSettingsPage}
-        />
-        <Navigator
-          {...this.state}
-          withTheme={this.withTheme}
-          openHomePage={this.openHomePage}
-          openSettingsPage={this.openSettingsPage}
-        />
-      </view>
-    );
-  }
+  return (
+    <view clip-radius="true" style={{ height: '100%' }}>
```

Comment:
@zhongyr is there a performance difference between inline styling via strings or objects?